cijoe/templates/perf_report/bench.jinja2.rst

```diff
@@ -0,0 +1,232 @@
+
+.. note:: **TODO**: Write some prose here.
+
+.. raw:: pdf
+
+   PageBreak
+
+Software
+--------
+
+.. note::
+   **TODO**: Populate this section using the artifacts produces and data collected by
+   **CIJOE**, that is, the output from ``cat /boot/cmdline``, ``uname``,
+   ``dmesg``, ``/etc/os-release``.
+
+.. raw:: pdf
+
+   PageBreak
+
+Hardware
+--------
+
+.. note::
+   **TODO**: Populate this section using the artifacts produces by CIJOE, that is, the ``lshw``.
+
+BIOS
+----
+
+.. note::
+   **TODO**: Add BIOS name and version and describe settings changed from the default.
+
+SSD Preconditioning
+-------------------
+
+We are interested in examing relative performance of
+storage-abstraction-layers, for this we are exclusively issuing reads. No
+writes are sent to the devices. This is to avoid the effects of device-side
+garbage-collection, write-buffer flushes etc.
+
+Thus, currently, no pre-conditioning is done since without issuing writes, we
+mitigate black-box effects of device-side logic.
+
+.. raw:: pdf
+
+   PageBreak
+
+Methodology
+===========
+
+Establish roof-line:
+
+* Maximum achievable IOPS rate possible with the given hardware.
+
+  - Hardware SPECs 8M IOPS
+
+* Maximum achievable IOPS rate of ``bdev_nvme``
+
+  - Established by measuring rates obtained with ``bdevperf``
+
+* Maximum achievable IOPS rate of ``io_uring`` and ``io_uring_cmd``
+
+  - Established by measuring rates achieved with ``t/io_uring``
+
+* Measure the IOPS rate of ``bdev_xnvme``, ``bdev_aio``, and ``bdev_uring``
+
+With all the above, we can observe:
+
+* Compare how far the different IO storage-paths are from the HW-roofline
+
+.. raw:: pdf
+
+   PageBreak
+
+Results
+=======
+
+.. raw:: pdf
+
+   PageBreak
+
+libaio
+------
+
+Comparing ``bdev_xnvme``, with ``io_mechanism=libaio``, to ``bdev_aio``.
+
+The line plot below displays the relationship between iodepth on the x-axis and
+IOPS (Input/Output Operations Per Second) on the y-axis, illustrating how IOPS
+performance varies with different iodepth values.
+
+.. image:: bdevperf_lineplot_libaio.png
+   :align: center
+   :width: 100%
+
+For ``libaio``, as shown above, the two lines overlap closely on the plot,
+indicating similar IOPS performance for both datasets across various iodepth
+values
+
+.. raw:: pdf
+
+   PageBreak
+
+io_uring
+--------
+
+Comparing ``bdev_xnvme``, using ``io_mechanism=io_uring``, to ``bdev_uring``.
+
+The line plot below displays the relationship between iodepth on the x-axis and
+IOPS (Input/Output Operations Per Second) on the y-axis, illustrating how IOPS
+performance varies with different iodepth values.
+
+.. image:: bdevperf_lineplot_io_uring.png
+   :align: center
+   :width: 100%
+
+For ``io_uring``, as shown above, The plot reveals that without IO-polling,
+then ``bdev_xnvme`` exhibits slightly lower IOPS than the reference
+implementation. However, when IO-polling is enabled, there's a noticeable boost
+in IOPS, showcasing its distinct advantage.
```

Comment:
Maybe it should be made more explicit that without IO-polling means with conserve cpu

Reply:
Good point, adding that.